tests/middlewares/test_auth.py

```diff
@@ -0,0 +1,77 @@
+"""
+    tests.middlewares.test_auth
+    ---------------------------
+
+    Tests XSnippet authentication middleware.
+
+    :copyright: (c) 2016 The XSnippet Team, see AUTHORS for details
+    :license: MIT, see LICENSE for details
+"""
+
+from aiohttp import web
+from jose import jwt
+import pytest
+import pkg_resources
+
+from xsnippet_api import application, conf
+from tests import AIOTestMeta, AIOTestApp
+
+
+class TestAuthMiddleware(metaclass=AIOTestMeta):
+
+    conf = conf.get_conf(
+        pkg_resources.resource_filename('xsnippet_api', 'default.conf'))
+    conf['auth'] = {'secret': 'SWORDFISH'}
+
+    def setup(self):
+        self.app = application.create_app(self.conf)
```

Comment:
I think it makes sense to create bare application instance since we ain't interested in XSnippet endpoints and so on. See how it's done in `tests/tests/test_resource.py`. In that case you won't even need to define and hijack custom resource.